From 101fa9f8893ecce519e7b43dbb7920473356f076 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sun, 10 Aug 2014 18:20:23 -0400
Subject: [iommu] Change iommu driver to call add_device_to_group trace event

Message-id: <20140810182023.24055.59464.stgit@gir.stowe>
Patchwork-id: 87580
O-Subject: [RHEL7.1 PATCH 09/19] iommu: Change iommu driver to call add_device_to_group trace event
Bugzilla: 1127467
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1127467
Upstream Status: d1cf7e822746b5e755f5a893ffeced1f6311c0cf

commit d1cf7e822746b5e755f5a893ffeced1f6311c0cf
Author: Shuah Khan <shuah.kh@samsung.com>
Date:   Thu Aug 15 11:59:24 2013 -0600

    iommu: Change iommu driver to call add_device_to_group trace event

    Change iommu driver to call add_device_to_group trace event. This iommu_grou
    class event can be enabled to trigger when devices get added to an iommu gro
    Trace information includes iommu group id and device name.

    Testing:
    The following is trace is generated when intel-iommu driver adds devices to
    to iommu groups during boot-time during its initialization:

           swapper/0-1     [003] ....     1.854793: add_device_to_group: IOMMU:
           swapper/0-1     [003] ....     1.854797: add_device_to_group: IOMMU:

    Signed-off-by: Shuah Khan <shuah.kh@samsung.com>
    Signed-off-by: Joerg Roedel <joro@8bytes.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/iommu.c b/drivers/iommu/iommu.c
index 58f6a16..349c92d 100644
--- a/drivers/iommu/iommu.c
+++ b/drivers/iommu/iommu.c
@@ -364,6 +364,8 @@ rename:
  /* Notify any listeners about change to group. */
  blocking_notifier_call_chain(&group->notifier,
          IOMMU_GROUP_NOTIFY_ADD_DEVICE, dev);
+
+ trace_add_device_to_group(group->id, dev);
  return 0;
 }
 EXPORT_SYMBOL_GPL(iommu_group_add_device);
-- 
1.7.1